src/HospitalLibrary/Rooms/EquipmentReallocationService.cs

```diff
@@ -0,0 +1,222 @@
+﻿using HospitalLibrary.Appointments;
+using HospitalLibrary.Migrations;
+using HospitalLibrary.Rooms;
+using HospitalLibrary.Rooms.DTOs;
+using HospitalLibrary.Rooms.Interfaces;
+using HospitalLibrary.Rooms.Model;
+using HospitalLibrary.Shared.Interfaces;
+using HospitalLibrary.Shared.Repository;
+using Microsoft.EntityFrameworkCore.Internal;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reactive.Subjects;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace HospitalLibrary.Rooms
+{
+    public class EquipmentReallocationService : IEquipmentReallocationService
+    {
+        private readonly ITimeIntervalValidationService _intervalValidation;
+        private readonly IUnitOfWork _unitOfWork;
+
+        public EquipmentReallocationService(IUnitOfWork unitOfWork,ITimeIntervalValidationService intervalValidation)
+        {
+            _unitOfWork = unitOfWork;
+            _intervalValidation = intervalValidation;
+        }
+
+        public async Task<EquipmentReallocation> Create(EquipmentReallocation equipmentReallocation)
+        {
+            await _intervalValidation.ValidateReallocation(equipmentReallocation);
+            _unitOfWork.EquipmentReallocationRepository.Add(equipmentReallocation);
+            _unitOfWork.EquipmentReallocationRepository.Save();
+
+            return equipmentReallocation;
+        }
+        public Task Delete(int id)
+        {
+            return null;
+        }
+
+        public Task<IEnumerable<EquipmentReallocation>> GetAll()
+        {
+            return null;
```

Comment:
Methods that have no implementation should not be in interface. 4th rule of SOLID